src/main/java/com/mongodb/spark/sql/connector/read/MongoScanBuilder.java

```diff
@@ -194,35 +197,44 @@ private FilterAndPipelineStage processFilter(final Filter filter) {
       EqualNullSafe equalNullSafe = (EqualNullSafe) filter;
       return new FilterAndPipelineStage(
           filter,
-          Filters.eq(
-              equalNullSafe.attribute(),
-              processValue(equalNullSafe.attribute(), equalNullSafe.value())));
+          getBsonValue(equalNullSafe.attribute(), equalNullSafe.value())
+              .map(bsonValue -> Filters.eq(equalNullSafe.attribute(), bsonValue))
+              .orElse(null));
     } else if (filter instanceof EqualTo) {
       EqualTo equalTo = (EqualTo) filter;
       return new FilterAndPipelineStage(
           filter,
-          Filters.eq(equalTo.attribute(), processValue(equalTo.attribute(), equalTo.value())));
+          getBsonValue(equalTo.attribute(), equalTo.value())
+              .map(bsonValue -> Filters.eq(equalTo.attribute(), bsonValue))
+              .orElse(null));
     } else if (filter instanceof GreaterThan) {
       GreaterThan greaterThan = (GreaterThan) filter;
       return new FilterAndPipelineStage(
           filter,
-          Filters.gt(
-              greaterThan.attribute(), processValue(greaterThan.attribute(), greaterThan.value())));
+          getBsonValue(greaterThan.attribute(), greaterThan.value())
+              .map(bsonValue -> Filters.gt(greaterThan.attribute(), bsonValue))
+              .orElse(null));
     } else if (filter instanceof GreaterThanOrEqual) {
       GreaterThanOrEqual greaterThanOrEqual = (GreaterThanOrEqual) filter;
       return new FilterAndPipelineStage(
           filter,
-          Filters.gte(
-              greaterThanOrEqual.attribute(),
-              processValue(greaterThanOrEqual.attribute(), greaterThanOrEqual.value())));
+          getBsonValue(greaterThanOrEqual.attribute(), greaterThanOrEqual.value())
+              .map(bsonValue -> Filters.gte(greaterThanOrEqual.attribute(), bsonValue))
+              .orElse(null));
     } else if (filter instanceof In) {
       In inFilter = (In) filter;
-      Bson pipelineStage =
-          Filters.in(
-              inFilter.attribute(),
-              Arrays.stream(inFilter.values())
-                  .map(v -> processValue(inFilter.attribute(), v))
-                  .collect(Collectors.toList()));
+      List<BsonValue> values =
```

Comment:
From what I can tell by running the modified test with a coverage tool, this branch is not tested (nor are most of the modified branches above).  Can the test coverage be improved?

Reply:
There is a test for all the filters: MongoSparkConnectorReadTest#testReadsAreSupportedWithFilters that remains unchanged.

Although I can't get the AND filter to be picked explicitly Spark, it coming as multiple Filters in DataSourceV2. But the logic for it seems sound so I don't want to remove it incase Spark gets updated in the future.

Reply:
My concern is that the modified test doesn't cover the modified code paths, which implies that there are no tests for the new behavior.   Sorry if I'm misunderstanding.

Reply:
The pipeline stage resulting is the same for the IN test we currently have. The extra logic is belt and braces as I don't know how to write an erroneous IN column that would force us to not be able to parse the BsonValue. 

That said sometimes users have used Scala types that have cause issues before. In the event of that happening I only want to push down the IN statement if all values can be converted to BsonValues - otherwise it would potentially miss data.

Reply:
OK, resolving.